library/src/webapp/js/sakai-message-broker.js

```diff
@@ -25,14 +25,15 @@ if (portal?.user?.id) {
       }
 
       if (differentUser) {
-        registration.pushManager.getSubscription().then(subscription => {
 
-          subscription.unsubscribe().then(successful => {
-
-            portal.notifications.subscribeIfPermitted(registration);
-          });
-        });
-      }
+        registration.pushManager.getSubscription()
+          .then(subscription => {
+            if (subscription) {
+                return subscription.unsubscribe();
+            }
+          })
+          .then(() => portal.notifications.subscribeIfPermitted(registration));
```

Comment:
I feel like this should be "finally" instead of "then". Don't we always want a subscription to happen?